aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,955 @@
+        let ethereum_address = ethereum::ethereum_address_no_eip55(recipient);  
+    
+        // Increment and retrieve the nonce  
+        let nonce = increment_and_get_nonce(initiator_address);  
+    
+        // Create bridge transfer details  
+        let details = native_bridge_store::create_details(  
+            initiator_address,  
+            ethereum_address, 
+            amount,  
+            nonce  
+        );  
+    
+        // Generate a unique bridge transfer ID  
+        // Todo: pass the nonce in here and modify the function to take a nonce. Or only use the nonce in native_bridge_store
+        let bridge_transfer_id = native_bridge_store::bridge_transfer_id(&details);  
+    
+        // Add the transfer details to storage  
+        native_bridge_store::add(bridge_transfer_id, details);
+    
+        // Push details to be able to lookup by nonce
+        native_bridge_store::set_nonce_to_bridge_transfer_id(nonce, bridge_transfer_id);
+
+        // Burn the amount from the initiator  
+        native_bridge_core::burn(initiator_address, amount);  
+    
+        let bridge_events = borrow_global_mut<BridgeEvents>(@aptos_framework);
+
+        // Emit an event with nonce  
+        event::emit_event(  
+             &mut bridge_events.bridge_transfer_initiated_events,
+            BridgeTransferInitiatedEvent {  
+                bridge_transfer_id,  
+                initiator: initiator_address,  
+                recipient,  
+                amount,  
+                nonce,  
+            }  
+        );  
+    }
+
+    /// Completes a bridge transfer by the initiator.
+    ///  
+    /// @param caller The signer representing the bridge operator.  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param bridge_transfer_id The unique identifier for the bridge transfer.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    /// @param nonce The unique nonce for the transfer.    
+    /// @abort If the caller is not the bridge operator or the transfer has already been processed.  
+    public entry fun complete_bridge_transfer(  
+        caller: &signer,  
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,  
+        recipient: address,  
+        amount: u64,  
+        nonce: u64  
+    ) acquires BridgeEvents {  
+        // Ensure the caller is the bridge operator
+        native_bridge_configuration::assert_is_caller_operator(caller);  
+
+        // Check if the bridge transfer ID is already associated with an incoming nonce
+        let incoming_nonce_exists = native_bridge_store::is_incoming_nonce_set(bridge_transfer_id);
+        assert!(!incoming_nonce_exists, ETRANSFER_ALREADY_PROCESSED); // Abort if the transfer is already processed
+
+        let ethereum_address = ethereum::ethereum_address_no_eip55(initiator);
```

Comment:
probably remove all eip55 references, this would be called ethereum_address or check_ethereum_address

Reply:
Right now there's still an option to either do an EIP 55 check or not. But it would be good to make a team decision about that.

Reply:
cool, I would do the check for hexadecimal digits only.

Is it something we should introduce on solidity as well?

Reply:
On Solidity won't it just revert automatically if there's a bad address? Not sure if complete_bridge_transfer has a check for the validity of the recipient on the Solidity side, that's probably worth looking into.

Reply:
I mean, checking if the input move address is okay.